src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -362,14 +393,20 @@ def execute(
         plan: SnowflakePlan,
         to_pandas: bool = False,
         to_iter: bool = False,
+        block: bool = True,
+        data_type: AsyncDataType = AsyncDataType.ROW,
         **kwargs,
     ) -> Union[
         List[Row], "pandas.DataFrame", Iterator[Row], Iterator["pandas.DataFrame"]
     ]:
+        if is_in_stored_procedure() and not block:
+            raise ValueError("can not use async and stored procedure at the same time")
```

Comment:
How about "async query isn't yet supported in stored procedure."

Reply:
sure, will do